tests/system/python/service/test_service_notification.py

```diff
@@ -0,0 +1,199 @@
+
+    def test_service(self, reset_and_start_foglamp, service_branch, foglamp_url, wait_time, retries, remove_directories):
+        self._configure_and_start_service(service_branch, foglamp_url, remove_directories)
+
+        retry_count = 0
+        # only 2 services is being up by default i.e core and storage
+        default_registry_count = 2
+        service_registry = default_registry_count
+        while service_registry != 3 and retry_count < retries:
+            svc = self._get_result(foglamp_url, '/foglamp/service')
+            service_registry = svc['services']
+            retry_count += 1
+            time.sleep(wait_time * 2)
+
+        if len(service_registry) == default_registry_count:
+            assert False, "Failed to start the {} service".format(SERVICE)
+
+        self._verify_service(foglamp_url, 'running')
+
+        self._verify_audit_log_entry(foglamp_url, '/foglamp/audit?source=NTFST')
+
+    def test_get_default_notification_plugins(self, foglamp_url, remove_directories):
+        remove_directories(os.environ['FOGLAMP_ROOT'] + '/plugins/notificationDelivery')
+        remove_directories(os.environ['FOGLAMP_ROOT'] + 'cmake_build/C/plugins/notificationDelivery')
+        jdoc = self._get_result(foglamp_url, '/foglamp/notification/plugin')
+        assert [] == jdoc['delivery']
+        assert 1 == len(jdoc['rules'])
+        assert NOTIFY_RULE == jdoc['rules'][0]['name']
+
+    # FIXME: Add name with some special character, Once FOGL-2434 is done
+    @pytest.mark.parametrize("data", [
+        {"name": "Test1", "description": "Test 1 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "retriggered"},
+        {"name": "Test2", "description": "Test 2 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "toggled"},
+        {"name": "Test3", "description": "Test 3 notification", "rule": NOTIFY_RULE, "channel": NOTIFY_PLUGIN, "enabled": "false", "notification_type": "one shot"}
+    ])
+    def test_create_notification_instances_with_default_rule_and_channel_python35(self, foglamp_url, notify_branch, data, remove_directories):
+        def _add_notification_instance(payload):
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("POST", '/foglamp/notification', json.dumps(payload))
+            r = conn.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert "Notification {} created successfully".format(payload['name']) == jdoc['result']
+
+        # FIXME: Handle in a better way; we need below code once for a test
+        if data['name'] == 'Test1':
+            self._install_notify_plugin(notify_branch)
+            remove_directories("/tmp/foglamp-notify-{}".format(NOTIFY_PLUGIN))
+
+        _add_notification_instance(data)
+
+    def test_delivery_rule_plugin_with_python35(self, foglamp_url):
```

Comment:
should not this be another test class for RulesPlugin etc.

Reply:
and for inbuilt, explicitly say the word `inbuilt` in test name

Reply:
There is no `inbuilt` delivery plugin; only rule plugin `inbuilt` exists as of now i.e OverMaxRule

Reply:
I meant for rules plugin itself. test_delivery_with_inbuilt_rule_plugin_and_notify_python35 or test_inbuilt_rule_plugin_with_python35_delivery ?